spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/secretmanager/SecretManagerPropertySource.java

```diff
@@ -87,43 +86,49 @@ public Object getProperty(String name) {
 	}
 
 	private static Map<String, Object> createSecretsPropertiesMap(
-			SecretManagerServiceClient client, String projectId, String secretsPrefix, String version, Map<String, String> versions) {
+			SecretManagerServiceClient client, String projectId, String secretsPrefix, String defaultVersion, Map<String, String> versions) {
 
 		ListSecretsPagedResponse response = client.listSecrets(ProjectName.of(projectId));
-
 		Map<String, Object> secretsMap = new HashMap<>();
 		for (Secret secret : response.iterateAll()) {
 			String secretId = extractSecretId(secret);
-			ByteString secretPayload = resolveSecretVersion(client, projectId, version, versions, secretId);
-			secretsMap.put(secretsPrefix + secretId, secretPayload);
+			ByteString secretPayload = getSecretPayload(client, projectId, secretId, defaultVersion, versions);
+			if (secretPayload != null) {
+				secretsMap.put(secretsPrefix + secretId, secretPayload);
+			}
 		}
 
 		return secretsMap;
 	}
 
-	private static ByteString resolveSecretVersion(SecretManagerServiceClient client, String projectId, String version, Map<String, String> versions, String secretId) {
-		if (!versions.isEmpty() && versions.containsKey(secretId)) {
-			String secretVersion = versions.get(secretId);
-			return getSecretPayload(client, projectId, secretId, secretVersion);
-		}
-		return getSecretPayload(client, projectId, secretId, resolveVersion(version));
-	}
-
-	private static String resolveVersion(String version) {
-		return StringUtils.hasText(version) ? version : LATEST_VERSION_STRING;
-	}
-
 	private static ByteString getSecretPayload(
-			SecretManagerServiceClient client, String projectId, String secretId, String version) {
+			SecretManagerServiceClient client,
+			String projectId,
+			String secretId,
+			String defaultVersion,
+			Map<String, String> versions) {
+		String version = versions.containsKey(secretId) ? versions.get(secretId) : defaultVersion;
 
 		SecretVersionName secretVersionName = SecretVersionName.newBuilder()
 				.setProject(projectId)
 				.setSecret(secretId)
 				.setSecretVersion(version)
 				.build();
 
-		AccessSecretVersionResponse response = client.accessSecretVersion(secretVersionName);
-		return response.getPayload().getData();
+		try {
+			AccessSecretVersionResponse response = client.accessSecretVersion(secretVersionName);
+			return response.getPayload().getData();
+		}
+		catch (Exception e) {
```

Comment:
Why not just catch `NotFoundException` here?

Reply:
Ooops forgot I could do this; done.